CMakeLists.txt

```diff
@@ -141,3 +141,8 @@ add_custom_target(update-headers
         VERBATIM
         COMMAND ${SHELL_FOR_PYTHON} -c "python ./generate_header.py ${CMAKE_SOURCE_DIR}/include/can/core/ids.hpp --source-dir=${OPENTRONS_HARDWARE_PATH} --language=c++"
         COMMAND ${SHELL_FOR_PYTHON} -c "python ./generate_header.py ${CMAKE_SOURCE_DIR}/include/bootloader/core/ids.h --source-dir=${OPENTRONS_HARDWARE_PATH} --language=c")
+
+add_custom_target(update-ot3-state-manager-headers
```

Comment:
finally, once you've done this (and this may not be a this-pr thing) it becomes a lot easier to bring the poetry stuff up into cmake rather than living in its own little makefile world. once you have a subdirectory like this, you can do the poetry install stuff in [https://cmake.org/cmake/help/latest/command/execute_process.html](execute_process) so it runs automatically.

Then, you can tell cmake that it must regenerate the project if the pyproject.toml` or `poetry.lock` files change by adding those files to [CMAKE_CONFIGURE_DEPENDS](https://cmake.org/cmake/help/latest/prop_dir/CMAKE_CONFIGURE_DEPENDS.html): `set_property(DIRECTORY APPEND PROPERTY CMAKE_CONFIGURE_DEPENDS pyproject.toml poetry.lock)`

At that point, you should see that when running `cmake --preset=simulators`, cmake runs poetry install; and that if you edit either file, running `cmake --build --preset=simulators` will rerun project generation, so you'll get any changed dependencies.

Reply:
> this isn't quite the right way to do this in cmake, for a couple reasons:
> 
> * The output gets emitted to the source directory. this requires a gitignore and is generally not great practice; you really want only stuff you check in to live in the source directory. That way, you can `rm` the build directory and be confident everything did get wiped out. I'd recommend doing the following:
>   
>   1. Put this command, and others related to `ot3_state_manager`, in `ot3_state_manager/CMakeLists.txt` (and `add_subdirectory` it, ideally guarded by `if(NOT ${CMAKE_CROSSCOMPILING})`
>   2. In that `CMakeLists.txt`, generate this header file to `${CMAKE_CURRENT_BINARY_DIR}`
>   3. use `add_library(ot3_state_manager INTERFACE)` and then `target_include_directories(ot3_state_manager INTERFACE ${CMAKE_CURRENT_BINARY_DIR}` to create a special header-only library that only exists to propagate this include
>   4. Add `target_link_libraries(${SIMEXE} ot3_state_manager)`, replacing `${SIMEXE}` with the name of the simulator executable, to each simulator executable in its cmakelists
> 
> By doing this, you get the following benefits:
> 
> 1. The simulator executables have a known-to-cmake explicit dependency on the generated header, which means if you update the header they should get rebuilt
> 2. you can get rid of the gitignore because the header is in the build dir

@sfoster1 I made the changes in https://github.com/Opentrons/ot3-firmware/pull/435/commits/083d1708763c5ac30b95a646983690a371635705 but I am having trouble testing if the header file is actually being created. 

I ran `cmake --preset=host-gcc10 .` then `cmake --build --preset=simulators` and I can't find the header file anywhere